src/whylogs/core/statistics/stringtracker.py

```diff
@@ -0,0 +1,245 @@
+import logging
+import math
+
+from datasketches import frequent_strings_sketch
+
+from whylogs.core.statistics.thetasketch import ThetaSketch
+from whylogs.core.summaryconverters import from_string_sketch
+from whylogs.proto import CharPosMessage, CharPosSummary, StringsMessage, StringsSummary
+from whylogs.util import dsketch
+from whylogs.util.util_functions import encode_to_integers
+
+from .numbertracker import NumberTracker
+
+MAX_ITEMS_SIZE = 128
+MAX_SUMMARY_ITEMS = 100
+
+logger = logging.getLogger(__name__)
+
+
+class CharPosTracker:
+    def __init__(self, character_list: str = "abcdefghijklmnopqrstuvwzyz0123456789-"):
+
+        self.character_list = character_list
+        self._character_list_map = {x: idx for idx, x in enumerate(character_list)}
+
+        self.char_pos_map = [NumberTracker() for _ in self.character_list]
+
+    def update(self, value):
+
+        for indx, char in enumerate(value):
+            if char in self.character_list:
```

Comment:
Also, we should track the unsupported characters as well

Reply:
Thats a good point, i will add that.

Reply:
done